### PR TITLE
shopinvader_search_engine: ease override of depends for category compute

### PR DESCRIPTION
**Pull request by cyrilmanuel**
ease override of depends for category compute to be override by other category fields.

### CODE REVIEW

**Comment by simahawk** [comment]
Not a "fix" :wink:  Also, can you please remove the odoo version from the commit?
Should be something like `shopinvader_search_engine: ease override of depends for category compute`

**Review by santostelmo — APPROVED** [review]

**Review by simahawk — APPROVED** [review]
LG

**Comment by cyrilmanuel** [comment]
hii @lmignon can you merge this PR ?

File: shopinvader_search_engine/models/product_template.py [hunk 1]
```diff
@@ -20,6 +20,9 @@ def _get_parent_categories(self, categ_ids):
         return categories
 
     @api.depends_context("index_id")
-    @api.depends("categ_id", "categ_id.parent_id")
+    @api.depends(lambda self: self._compute_shopinvader_category_depends())
```
Comment (lmignon) [inline shopinvader_search_engine/models/product_template.py:23]: To extend the `depends` you just need to override the method with a new `api.depends` decorator with only your new dependent. The list of depends is computed by collecting all the depends declared across the class hierarchy. see (https://github.com/odoo/odoo/blob/16.0/odoo/fields.py#L568)

**Review by lmignon — CHANGES_REQUESTED** [review]

File: shopinvader_search_engine/models/product_template.py (hunk 1 above)
Comment (simahawk) [reply shopinvader_search_engine/models/product_template.py:23]: Yes but I find this way more explicit and flexible. It's also a well known pattern in the Odoo ecosystem. That's why I proposed it.

Comment (lmignon) [reply shopinvader_search_engine/models/product_template.py:23]: It depends... Both are valid. The `override` approach has the advantage of allowing you to easily find all `depends` for a given field using a regular expression...

Comment (simahawk) [reply shopinvader_search_engine/models/product_template.py:23]: > It depends... Both are valid. The `override` approach has the advantage of allowing you to easily find all `depends` for a given field using a regular expression...

You can find these overrides way easier IMO.

The final question is: is it really a blocking point or not? :)

Comment (lmignon) [reply shopinvader_search_engine/models/product_template.py:23]: The final question is 'What guidelines do we want to promote to ensure that the same patterns are used across all shopinvader modules, and to ensure consistency?' @qgroulard @AnizR @marielejeune @sebastienbeau

Comment (AnizR) [reply shopinvader_search_engine/models/product_template.py:23]: I agree with @lmignon, both approaches are valid but I personally prefer the 'override' approach with new @api.depends

**Review by benwillig — APPROVED** [review]

File: shopinvader_search_engine/models/product_template.py (hunk 1 above)
Comment (sebastienbeau) [reply shopinvader_search_engine/models/product_template.py:23]: using @api.depends() is enough, I do not see case that are not solve with this solution.

**Comment by cyrilmanuel** [comment]
hi @lmignon i think we have 50 50. can we merge it to go ahead please ?

**Comment by lmignon** [comment]
> hi @lmignon i think we have 50 50. can we merge it to go ahead please ?

We'll keep the native approach and not not merge this one. I'm don't know how Odoo will work if both approach are used at same time and I would avoid to constrain everyone to use this specific one and have to know it must override a specific method.

**Comment by sebastienbeau** [comment]
I close it as using @api.depends() is enough, I do not see case that are not solve with this solution.

**Pull request closed without merging**